### PR TITLE
Enter splits lines in the middle in insert mode

### PR DESCRIPTION
**Pull request by vanceza**

### CODE REVIEW

**Comment by WuTheFWasThat** [comment]
nvm, github's UI confused me

**Comment by vanceza** [comment]
Updated, but I think the original code was cleaner. Maybe I should add support for multiple registers, even if we don't expose it to the user?

**Comment by WuTheFWasThat** [comment]
I'm not opposed to that

File: assets/js/register.coffee [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Register
-  TYPES = {
+  this.TYPES = {
```
Comment (WuTheFWasThat) [inline assets/js/register.coffee]: @TYPES

File: assets/js/register.coffee [hunk 2]
```diff
@@ -7,24 +7,41 @@ class Register
...
     @chars = chars
 
   saveRows: (serialized_rows) ->
-    @type = TYPES.ROWS
+    @type = Register.TYPES.ROWS
     @serialized_rows = serialized_rows
+  
+  serialize: () ->
+    data = ''
+    if @type==Register.TYPES.CHARS
```
Comment (WuTheFWasThat) [inline assets/js/register.coffee]: i mildly prefer spaces around the double equals

**Comment by WuTheFWasThat** [comment]
looks good overall.  fix nits, add a few more test cases, and then you can rebase+squash and push.

test case ideas:
- edge cases, i.e. cursor is at beginning or end
- line being split has children

File: assets/js/register.coffee (hunk 2 above)
Comment (vanceza) [inline assets/js/register.coffee]: My usual take on style stuff is that I'll keep it in mind for future commits but don't want it to block progress. If you set up a linting system and a git hook that calls it, I would use that.
